WordPress/Classes/ViewRelated/Suggestions/SuggestionsTableView.swift

```diff
@@ -21,26 +21,40 @@ extension SuggestionType {
     /// Show suggestions for the given word.
     /// - Parameters:
     ///   - word: Used to find the suggestions that contain this word.
+    ///   - completionHandler: The completion handler takes a boolean that's true when at least one suggestion is being shown.
     /// - Returns: True when at least one suggestion is being shown.
-    @discardableResult func showSuggestions(forWord word: String) -> Bool {
-        guard self.enabled else { return false }
-
-        if word.hasPrefix(suggestionTrigger) {
-            let searchQuery = NSString(string: word).substring(from: suggestionTrigger.count)
-            self.searchText = word
-            self.searchResults = self.searchResults(
-                searchQuery: searchQuery,
-                suggestions: suggestions ?? [],
-                suggestionType: suggestionType
-            )
-        } else {
-            self.searchText = ""
-            self.searchResults = []
+    func showSuggestions(forWord word: String, completionHandler: ShowSuggestionsCompletionHandler? = nil) {
+        guard self.enabled else {
+            completionHandler?(false)
+            return
         }
 
-        self.tableView.reloadData()
-        self.setNeedsUpdateConstraints()
-        return searchResults.count > 0
+        // Cancel previous search operation
+        self.searchOperationQueue.cancelAllOperations()
+
+        // Perform search in a background thread
+        self.searchOperationQueue.addOperation { [weak self] in
```

Comment:
👋 @salimbraksa - do you have a recommendation on how I could stress test this and determine when an operation has been cancelled?

I'm not fluent in NSOperationQueues, but the docs make it sound like we'd still have to do a [check in the operation ](https://developer.apple.com/documentation/foundation/nsoperationqueue/1417849-cancelalloperations) to determine whether to proceed or stop, which I don't see.

Thanks!